typing/typecore.ml

```diff
@@ -428,6 +456,7 @@ let enter_variable ?(is_module=false) ?(is_as_variable=false) loc name mode ty
     (* Note: unpack patterns enter a variable of the same name *)
     if not !allow_modules then
       raise (Error (loc, Env.empty, Modules_not_allowed));
+    escape ~loc ~env:Env.empty mode;
```

Comment:
I don't understand this check.

Reply:
If you unpack a first-class module then it should be heap, since we do not track whether modules escape.